src/microbe_stage/MicrobeAI.cs

```diff
@@ -38,13 +39,26 @@ public class MicrobeAI
     [JsonProperty]
     private float pursuitThreshold;
 
+    /// <remarks>
+    ///   Memory of the previous absorption step is required to compute gradient.
+    ///   Values dictionary rather than single value as they will be combined with variable weights.
+    /// </remarks>
+    [JsonProperty]
+    private Dictionary<Compound, float> previouslyAbsorbedCompounds;
```

Comment:
Am I missing something obvious?

This variable gets declared here.
It gets initialized on line 60 with the values of `TotalAbsorbedCompounds`.
Lines 82 to 88 also apply the values of `TotalAbsorbedCompounds` to this variable.
Line 404 reads the value.

Why don't you just access `TotalAbsorbedCompounds` directly on line 404?

Reply:
I guess this is a tricky part that definitely seems to need some more comments (which I've just written): lines 82->88 store the value from `TotalAbsorbedCompounds`, and line 89 clears `TotalAbsorbedCompounds` just afterwards (to free it the new absorption values for next time step). So previouslyAbsorbedCompounds serves as a storage (which is why it is always getting the value of `TotalAbsorbedCompounds`, but from the previous step).